src/annotator/anchoring/pdf.js

```diff
@@ -1,44 +1,75 @@
 /* global PDFViewerApplication */
 
-import seek from 'dom-seek';
-
 import RenderingStates from '../pdfjs-rendering-states';
 
-import { BrowserRange } from './range';
-import { toRange as textPositionToRange } from './text-position';
-import { TextPositionAnchor, TextQuoteAnchor } from './types';
+import { TextPosition, TextRange } from './text-range';
+import { TextQuoteAnchor } from './types';
 
 /**
  * @typedef {import('../../types/api').TextPositionSelector} TextPositionSelector
  * @typedef {import('../../types/api').TextQuoteSelector} TextQuoteSelector
+ * @typedef {TextPositionSelector|TextQuoteSelector} Selector
  *
  * @typedef {import('../../types/pdfjs').PDFPageView} PDFPageView
  * @typedef {import('../../types/pdfjs').PDFViewer} PDFViewer
  */
 
+/**
+ * @typedef PdfTextRange
+ * @prop {number} pageIndex
+ * @prop {Object} anchor
+ * @prop {number} anchor.start
+ * @prop {number} anchor.end
+ */
+
 // Caches for performance.
 
 /**
- * Map of page index to page text content as a `Promise<string>`.
+ * Map of page index to page text content.
+ *
+ * @type {Record<number,Promise<string>>}
  */
 let pageTextCache = {};
 
 /**
- * 2D map from `[quote][position]` to `{pageIndex, anchor}` intended to optimize
- * re-anchoring of a pair of quote and position selectors if the position
- * selector fails to anchor on its own.
+ * A cache that associates the quote and position from an annotation's
+ * selectors to the resolved location in the PDF.
+ *
+ * This helps with re-anchoring of a pair of quote and position selectors if
+ * the position selector fails to anchor.
+ *
+ * @type {Record<string, Record<number, PdfTextRange>>}
  */
 let quotePositionCache = {};
 
+/**
+ * Return offset of `node` among its siblings
+ *
+ * @param {Node} node
+ */
 function getSiblingIndex(node) {
-  return Array.from(node.parentNode.childNodes).indexOf(node);
+  let index = 0;
+  while (node.previousSibling) {
+    ++index;
+    node = node.previousSibling;
+  }
+  return index;
+}
```

Comment:
The change in `getSiblingIndex` resolves a dubious technicality flagged by TypeScript that it would throw if passed a node with no parent.